src/masternode-payments.cpp

```diff
@@ -777,6 +779,62 @@ bool CMasternodePayments::ProcessBlock(int nBlockHeight, CConnman& connman)
     return false;
 }
 
+void CMasternodePayments::CheckPreviousBlockVotes(int nPrevBlockHeight)
+{
+    LogPrint("mnpayments", "CMasternodePayments::CheckPreviousBlockVotes -- nPrevBlockHeight=%d, expected voting MNs:\n",
+             nPrevBlockHeight);
```

Comment:
A set of `LogPrint`/`LogPrintf`-s isn't guaranteed to execute in a non-interrupted sequence without some global lock, other threads might also write to the log and if we are going to write a lot of individual lines to the logfile reading this function output can become pretty hard. To avoid that you could compile the resulting string first and dump it to the log only when it's ready i.e. smth like
```
std::string strLog = strprintf("CMasternodePayments::CheckPreviousBlockVotes -- nPrevBlockHeight=%d, expected voting MNs:\n", nPrevBlockHeight);
...
strLog += strprintf("CMasternodePayments::CheckPreviousBlockVotes --   could not find vote %s\n", voteHash.ToString());
...
...//same for other cases...
...
LogPrintf("%s", strLog);
```

Reply:
Implemented suggestion